File: gin-vue-admin/server/router/ffproduction/ff_operation_assignment.go
package ffproduction

import (
	"gin-vue-admin/server/middleware"
	"github.com/gin-gonic/gin"
)

type OperationAssignmentRouter struct{}

func (e *OperationAssignmentRouter) InitOperationAssignmentRouter(Router *gin.RouterGroup) {
	assignmentRouter := Router.Group("ffProduct").Use(middleware.OperationRecord())
	assignmentRouterWithoutRecord := Router.Group("ffProduct")
	{
		assignmentRouter.POST("createOperationAssignment", operationAssignmentApi.CreateOperationAssignment)
		assignmentRouter.PUT("updateOperationAssignment", operationAssignmentApi.UpdateOperationAssignment)
		assignmentRouter.DELETE("deleteOperationAssignment", operationAssignmentApi.DeleteOperationAssignment)
	}
	{
		assignmentRouterWithoutRecord.GET("getOperationAssignmentList", operationAssignmentApi.GetOperationAssignmentList)
	}
}

File: gin-vue-admin/web/src/api/ffOperationAssignment.js
import service from '@/utils/request'

export const createOperationAssignment = (data) => {
  return service({
    url: '/ffProduct/createOperationAssignment',
    method: 'post',
    data
  })
}

export const updateOperationAssignment = (data) => {
  return service({
    url: '/ffProduct/updateOperationAssignment',
    method: 'put',
    data
  })
}

export const deleteOperationAssignment = (data) => {
  return service({
    url: '/ffProduct/deleteOperationAssignment',
    method: 'delete',
    data
  })
}

export const getOperationAssignmentList = (params) => {
  return service({
    url: '/ffProduct/getOperationAssignmentList',
    method: 'get',
    params
  })
}
